nncf/parameters.py

```diff
@@ -56,3 +56,17 @@ class DropType(Enum):
 
     ABSOLUTE = "absolute"
     RELATIVE = "relative"
+
+
+@api(canonical_alias="nncf.CompressWeightsMode")
+class CompressWeightsMode(Enum):
+    """
+    Defines a mode for weight compression.
+
+    :param COMPRESSED_INT8: Stands for int8 quantization of all weights.
+    :param COMPRESSED_NF4: assumes mixed precision quantization, when first and last layers are always compressed to
+        INT8, and all others are quantized whether to NF4 or to INT8 depending on some criteria and the given ratio.
+    """
+
+    COMPRESSED_INT8 = "compressed_int8"
+    COMPRESSED_NF4 = "compressed_nf4"
```

Comment:
`COMPRESSED_NF4` is mixed mode. Should we highlight mixed precision in the name? For example,  `COMPRESSED_NF4_INT8`

Reply:
@AlexKoff88 please comment

Reply:
It can be mixed with different precision, but NF4 is favorable precision in this case. By default, all layers is NF4, except last and first. Decided to keep this name and explain details in the docstring.

Reply:
Here we define the baseline compression type. Actually, the model get FP32/FP16-INT8-NF4 and the ratio may vary. I think it is enough to add a comment saying that the model is mixed precision. To tell the truth, all our models are mixed precision as you know.